x-pack/docs/en/security/securing-communications/update-tls-certificates.asciidoc

```diff
@@ -0,0 +1,784 @@
+
+   j. Repeat these steps for each additional node in your cluster.
+
+. After generating a certificate for each of your nodes, enter a password for
+   your keystore when prompted.
+
+. Unzip the generated `elasticsearch-ssl-http.zip` file. This compressed file
+   contains one directory for both {es} and {kib}. Within the `/elasticsearch`
+   directory is a directory for each node that you specified with it's own
+   `http.p12` file. For example:
++
+--
+[source,txt]
+----
+/node1
+|_ README.txt
+|_ http.p12
+|_ sample-elasticsearch.yml
+----
+
+[source,txt]
+----
+/node2
+|_ README.txt
+|_ http.p12
+|_ sample-elasticsearch.yml
+----
+
+[source,txt]
+----
+/node3
+|_ README.txt
+|_ http.p12
+|_ sample-elasticsearch.yml
+----
+--
+
+. If necessary, rename each `http.p12` file to match the name of your existing
+certificate for HTTP client communications. This keystore contains both a
+certificate and key.
+
+. [[start-rolling-restart-http-newca,step 5]]On the current node in your cluster where you're updating the keystore,
+start a <<restart-cluster-rolling,rolling restart>>.
++
+Stop at the step indicating *Perform any needed changes*, and then proceed to
+the next step in this procedure.
+
+. Replace your existing keystore with the new keystore, ensuring that the
+file names match. For example, `node1-http.p12`.
++
+If your 
+<<cert-password-updates,keystore password is changing>>, then save the
+keystore with a new filename so that {es} doesn't attempt to reload the file
+before you update the password.
+
+. If you needed to save the new keystore with a new filename, update the
+`ES_PATH_CONF/elasticsearch.yml` file to use the filename of the new keystore.
+For example:
++
+[source,yaml]
+----
+xpack.security.http.ssl.enabled: true
+xpack.security.http.ssl.keystore.path: node1-http.p12
+----
+
+. If your keystore password is changing, add the password for your private key
+to the secure settings in {es}.
++
+[source,shell]
+----
+./bin/elasticsearch-keystore add xpack.security.http.ssl.keystore.secure_password
+----
+
+. Start the node where you updated the keystore, and then complete the remaining
+steps for a <<restart-cluster-rolling,rolling restart>>.
+
+. [[verify-keystore-http-newca,step 10]](Optional) Use the <<security-api-ssl,SSL certificate API>> to verify that
+{es} loaded the new keystore.
+
+. One node at a time, complete <<start-rolling-restart-http-newca>> through
+<<verify-keystore-http-newca>> until you've updated all keystores in your cluster.
+
+. (Optional) After rotating keystores on each node in your cluster, remove the
+old CA certificate from your truststore. 
++
+[source,shell]
+----
+keytool -delete -noprompt -alias <old-ca>  -keystore config/http.p12 \
+-storepass <password>
+----
++
+[%collapsible%open]
+.Command parameters
+====
+   `-alias`:: Name of the old CA certificate that you want to remove from your
+   truststore.
+====
+
+[discrete]
+[[http-kibana-newca-whatsnext]]
```

Comment:
I think we need to add a note for this (kibana and other clients) somewhere before. If you have reached this point, then you have changed certificates successfully and in that, you have broken communication from kibana, ingestion from logstash and beats and agent etc.

Reply:
See [these changes](https://github.com/elastic/elasticsearch/pull/73781/commits/e7d031f4e6a04130326d49d0b3e601449478fa58#diff-1490c74621f912bbdeb341055e902df727b240257b2fbd22bf774da6f8332f70R523-R531), which I added to address this scenario and your [other comment](https://github.com/elastic/elasticsearch/pull/73781#discussion_r666499084) about updating clients beforehand.